Extensions/AboutUsParser.js

```diff
@@ -0,0 +1,267 @@
+});
+
+
+class Paragraph {
+    /**
+     * 
+     * @param {Object} object 
+     */
+    constructor(object) {
+        this.title = object.title;
+        this.short = object.text ? object.text.short : object.short;
+        this.long = object.text ? object.text.long : object.long;
+        this.language = object.language;
+        this.folder = object.folder;
+        this.photo = object.photo ? object.photo : null;
+    }
+
+    getDefaultHtmlCode() {
+            return `<div class="card about">
+        <div class="card-img">
+            ${this.photo ? `<img src="{fill_parents}${this.photo}" class="card-img-top" alt="Picture Not Available!">` : ''}
+        </div>
+        <div class="card-content">
+            <div>
+                <div class="card-title">
+                    <h3>${this.title}</h3>
+                </div>
+                <div class="card-text short">
+                    <p>${this.short}</p>
+                </div>
+            </div>
+            <div class="card-body">
+                <div class="card-text long">
+                ${this.long.map(photo =>`<p>${photo}</p>`).join('<br>')}
+                </div>
+            </div>
+            <div class="card-footer">
+                <button class="toggle-btn">...see more</button>
+            </div>
+        </div>
+    </div>`.trim();
+    }
+}
+
+function Files(path = null, result) {
+    path = path ? path : C.dPath;
+
+    getFiles(Path.join(path), (err, res) => {
+        if (err) throw new Error(err);
+        result(res.filter((file) => !fs.statSync(Path.join(file)).isDirectory()));
+    });
+}
+
+/**
+ * 
+ * @param {Array} files Array of files returned by function `Files(path, result)`
+ */
+async function createFilesArray(files) {
+    let mappedFiles = [];
+    files.map(file => {
+        file = file.split('/');
+        mappedFiles.push({ name: file[file.length - 1], path: Path.join(...file), language: file[file.length - 2], folder: file[file.length - 3] });
+    });
+    return mappedFiles;
+}
+
+/** Parses given string by separators and empty lines
+ * 
+ * @param {String} string text to be parsed
+ * @param {Function} data
+ * @returns {Array} 
+ */
+function parseString(string) {
+    try {
+        let sections = string.split(C.separator); // We split the string by our main separator into individual sections
+        let parsedSections = [];
+
+        sections.map(section => { // We iterate through every parsed section
+            let para = section.split('\r\n').filter(sentence => sentence.trim() ? true : false); // Splitting sections into paragraphs by new line and removing empty lines
+            //console.log(para.slice(2, para.length));
+            parsedSections.push({ title: para[0], short: para[1], long: para.slice(2,para.length) }); // Pushing a parsed section object into an array
+        });
+
+        return parsedSections; // return array of parsed sections as a callback
+    } catch (error) {
+        console.error(error);
```

Comment:
why are we catching the errors? shouldn't they break the script instead?

Reply:
Sorry, an old habit from other languages.